native/src/main/java/io/ballerina/stdlib/mime/util/EntityBodyHandler.java

```diff
@@ -442,31 +432,35 @@ public static void writeEventStreamToOutputStream(Environment env, BObject entit
     }
 
     private static void writeEvent(Environment env, BObject eventStreamWriter) {
-        env.getRuntime().invokeMethodAsyncConcurrently(eventStreamWriter, WRITE_EVENT_STREAM_METHOD, null, null,
-                new Callback() {
-                    @Override
-                    public void notifySuccess(Object result) {
-                        BObject entity = (BObject) eventStreamWriter.getNativeData(ENTITY);
-                        OutputStream outputStream = (OutputStream) eventStreamWriter.getNativeData(OUTPUT_STREAM);
-                        if (result == null) {
-                            entity.addNativeData(ENTITY_BYTE_STREAM, null);
-                            EntityBodyHandler.closeMessageOutputStream(outputStream);
-                            return;
-                        }
-                        if (result instanceof BError error) {
-                            entity.addNativeData(ENTITY_BYTE_STREAM, null);
-                            throw error;
-                        }
-                    }
-
-                    @Override
-                    public void notifyFailure(BError bError) {
-                        OutputStream outputStream = (OutputStream) eventStreamWriter.getNativeData(OUTPUT_STREAM);
-                        EntityBodyHandler.closeMessageOutputStream(outputStream);
-                        throw ErrorCreator.createError(StringUtils.fromString(
-                                "Error occurred while streaming content: " + bError.getMessage()));
-                    }
-                }, null, null);
+        try {
+            handleEventPanic(eventStreamWriter, env.getRuntime().callMethod(eventStreamWriter,
```

Comment:
Why this function is named as `handleEventPanic`? Isn't this a success scenario?

Reply:
this seems like a bug. I will update it